### PR TITLE
Detect Carbanak Step 2.B Screen Capture (Powershell)

### PR DESCRIPTION
**Pull request by fabamatic**
|Related issue|
|---|
|closes https://github.com/wazuh/wazuh/issues/9045|

## Description

This detection is based on PowerShell Script Block logs. Detects if a call to CopyFromScreen function was done by a PowerShell script.
Also set base rules for Powershell/Operational channel
 

## Configuration options
Powershell Script Block must be enabled on endpoint.
Wazuh client should be configured to forward Powershell/Operational logs to Wazuh manager

## Logs/Alerts example
Sample log:
```
{"win":{"eventdata":{"path":"C:\\\\Users\\\\AtomicRed\\\\AppData\\\\Roaming\\\\TransbaseOdbcDriver\\\\screenshot__.ps1","messageNumber":"1","messageTotal":"1","scriptBlockText":"function screenshot([Drawing.Rectangle]$bounds, $path){      $bmp = New-Object System.Drawing.Bitmap($bounds.width, $bounds.height)     $graphics = [Drawing.Graphics]::FromImage($bmp)     $graphics.CopyFromScreen($bounds.Location, [Drawing.Point]::Empty, $bounds.size)     $bmp.Save($path)     $graphics.Dispose()     $bmp.Dispose() }","scriptBlockId":"1b79ef82-0efe-4734-b092-3a4b9f17e080"},"system":{"eventID":"4104","keywords":"0x0","providerGuid":"{a0c1853b-5c40-4b15-8766-3cf1c58f985a}","level":"5","channel":"Microsoft-Windows-PowerShell/Operational","opcode":"15","message":"\"Creating Scriptblock text (1 of 1):\r\nfunction screenshot([Drawing.Rectangle]$bounds, $path){ \n    $bmp = New-Object System.Drawing.Bitmap($bounds.width, $bounds.height)\n    $graphics = [Drawing.Graphics]::FromImage($bmp)\n    $graphics.CopyFromScreen($bounds.Location, [Drawing.Point]::Empty, $bounds.size)\n    $bmp.Save($path)\n    $graphics.Dispose()\n    $bmp.Dispose()\n}\r\n\r\nScriptBlock ID: 1b79ef82-0efe-4734-b092-3a4b9f17e080\r\nPath: C:\\Users\\AtomicRed\\AppData\\Roaming\\TransbaseOdbcDriver\\screenshot__.ps1\"","version":"1","systemTime":"2021-06-17T19:42:48.3171903Z","eventRecordID":"95916","threadID":"5292","computer":"hrmanager.ExchangeTest.com","task":"2","processID":"1756","severityValue":"VERBOSE","providerName":"Microsoft-Windows-PowerShell"}}}
```

logtest output:
```
**Phase 1: Completed pre-decoding.

**Phase 2: Completed decoding.
        name: 'json'
        win.eventdata.messageNumber: '1'
        win.eventdata.messageTotal: '1'
        win.eventdata.path: 'C:\\Users\\AtomicRed\\AppData\\Roaming\\TransbaseOdbcDriver\\screenshot__.ps1'
        win.eventdata.scriptBlockId: '1b79ef82-0efe-4734-b092-3a4b9f17e080'
        win.eventdata.scriptBlockText: 'function screenshot([Drawing.Rectangle]$bounds, $path){      $bmp = New-Object System.Drawing.Bitmap($bounds.width, $bounds.height)     $graphics = [Drawing.Graphics]::FromImage($bmp)     $graphics.CopyFromScreen($bounds.Location, [Drawing.Point]::Empty, $bounds.size)     $bmp.Save($path)     $graphics.Dispose()     $bmp.Dispose() }'
        win.system.channel: 'Microsoft-Windows-PowerShell/Operational'
        win.system.computer: 'hrmanager.ExchangeTest.com'
        win.system.eventID: '4104'
        win.system.eventRecordID: '95916'
        win.system.keywords: '0x0'
        win.system.level: '5'
        win.system.message: '"Creating Scriptblock text (1 of 1):
function screenshot([Drawing.Rectangle]$bounds, $path){
    $bmp = New-Object System.Drawing.Bitmap($bounds.width, $bounds.height)
    $graphics = [Drawing.Graphics]::FromImage($bmp)
    $graphics.CopyFromScreen($bounds.Location, [Drawing.Point]::Empty, $bounds.size)
    $bmp.Save($path)
    $graphics.Dispose()
    $bmp.Dispose()
}

ScriptBlock ID: 1b79ef82-0efe-4734-b092-3a4b9f17e080
Path: C:\Users\AtomicRed\AppData\Roaming\TransbaseOdbcDriver\screenshot__.ps1"'
        win.system.opcode: '15'
        win.system.processID: '1756'
        win.system.providerGuid: '{a0c1853b-5c40-4b15-8766-3cf1c58f985a}'
        win.system.providerName: 'Microsoft-Windows-PowerShell'
        win.system.severityValue: 'VERBOSE'
        win.system.systemTime: '2021-06-17T19:42:48.3171903Z'
        win.system.task: '2'
        win.system.threadID: '5292'
        win.system.version: '1'

**Phase 3: Completed filtering (rules).
        id: '93002'
        level: '0'
        description: 'Screen capture method invoked from PowerShell script'
        groups: '['windows', ' powershell']'
        firedtimes: '1'
        mail: 'False'
        mitre.id: '['T1113']'
        mitre.tactic: '['Collection']'
        mitre.technique: '['Screen Capture']'

```

### CODE REVIEW

**Review by 72nomada — DISMISSED** [review]
LGTM

File: ruleset/rules/0915-win-powershell_rules.xml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    <rule id="91603" level="14">
+        <if_sid>91602</if_sid>
+        <field name="win.system.message">CopyFromScreen</field>
```
[the remark this answers is not in the document]
Comment (JavierBejMen) [reply ruleset/rules/0915-win-powershell_rules.xml:31]: Yes, we should use pcre2 as is going to be our next standard

File: ruleset/rules/0915-win-powershell_rules.xml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        <field name="win.system.channel">^Microsoft-Windows-PowerShell/Operational$</field>
+        <options>no_full_log</options>
+        <description>Group of Windows rules for the Powershell/Operational channel</description>
+    </rule>
+
+    <!-- Powershell Script Block rules -->
+
+    <rule id="91602" level="0">
+        <if_sid>91601</if_sid>
+        <field name="win.eventdata.ScriptBlockId">\.+</field>
```
Comment (JavierBejMen) [inline ruleset/rules/0915-win-powershell_rules.xml:23]:
```suggestion
        <field name="win.eventdata.ScriptBlockId" type="pcre2">.+</field>
```

**Review by JavierBejMen — CHANGES_REQUESTED** [review]
Use regex type PCRE2 when using patterns, otherwise LGTM

File: ruleset/rules/0915-win-powershell_rules.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (fabamatic) [reply ruleset/rules/0915-win-powershell_rules.xml:31]: Agree on the pcre2 standard.
The content of the field is win.system.message is a block of code, I search for the function name.

**Review by JavierBejMen — APPROVED** [review]

**Review by jcruzlp — APPROVED** [review]
LGTM!

**Pull request merged**